umklapp/views.py

```diff
@@ -160,9 +160,14 @@ def show_story(request, story_id):
     if not s.is_finished or (not s.participates_in(request.user) and not s.is_public):
         raise PermissionDenied
 
+    anonym = True
+    for t in s.tellers.all():
+        if t.id == request.user.id:
```

Comment:
I can see where the problem is: You compare a teller with a user. Surely this always returns false, since tellers are instances of Teller which is another type than User.
I don't think that your fix works because Tellers' ids are a different domain than Users' ids, so t.id will equal request.user.id only by coincidence...